### DISCUSSION TITLE
Resuming move after SSH session aborted while moving Synology Drive Server pkg w/ 600GB @synologydrive folder

### DISCUSSION THREAD

**Discussion by TorbenSchreiter** [007revad/Synology_app_mover#126] [Q&A] [answered]
Hi Dave,

thanks a lot for your community efforts particularly wrt Synology_app_mover & Synology_HDD_db! Have just donated.

Regardless of that: I am a bit late to the party to migrate to btrfs, which is the reason why over the years, the @synologydrive folder has grown dramatically to >600GB in size. Unfortunately, while moving the package from /volume1 to /volume2 the SSH session got reset during the last(?) step - while moving that folder. The move was only partially completed.

Both the DSM Package mgr and Synology_app_mover already locate the package at the new location. However there are now @synologydrive folders on both volumes.

I have tried to naively move the remainder portion like this, but it fails:
```
user123@GLHV3:/volume2$ cd @synologydrive
-sh: cd: @synologydrive: Permission denied
user123@GLHV3:/volume2$ sudo mv "/volume1/@synologydrive" "/volume2/"
Password:
mv: inter-device move failed: '/volume1/@synologydrive' to '/volume2/@synologydrive'; unable to remove target: Directory not empty
```

There is no backup of the pkg (due to the sheer size of the package).

Could you maybe help out with the best way how to move the remainder. Also not sure whether there are potentially other parts additionally to @synologydrive that also need to be manually moved now.

Thanks!

**Comment by 007revad** [comment]
I saw the donation. Thank you.

In Package Center, stop the Synology Drive Server package.

After moving `@synologydrive` it would move `@SynologyDriveShareSync` then edit 3 files and a symlink.

Give me an hour and  I'll work out the commands needed to fix `@synologydrive` and do the remaining steps.

**Reply by 007revad** [reply]
This should fix it. https://github.com/007revad/Synology_HDD_db/blob/test/app_mover_issue_126.zip Click on the "down arrow" button on the far right (that says "Download raw file" when you hover your mouse over the button).

1. Stop Synology Drive.
2. Run the issue.sh script. It may take a long while.

**Reply by TorbenSchreiter** [reply]
Thanks a lot. The script did eventually complete. However, both folders were not deleted from the source volume by the script. I therefore had them renamed appending ".old" before starting up the Synology Drive package again.

Unfortunately, it seems that the db on the new /volume2 is not actually found by the package. All settings inside Synology Drive Admin Console are virgin as with a fresh install.

Also at the same time on the old /volume1 a new `@synologydrive` was created (see below).

_in /volume1/:_
```
drwxr-xr-x   5 root       root           4096 Oct 17 12:30  @synologydrive
drwxr-xr-x   6 root       root           4096 Aug 16 08:35  @synologydrive.old
drwxr-xr-x   7 root       root           4096 Apr 14  2022  @SynologyDriveShareSync.old
```

Any thoughts?

**Reply by 007revad** [reply]
We'll need to check which volumes the symlinks are pointing to.

It's 10pm here. I'll get back to you in the morning.

**Reply by 007revad** [reply]
What do each of these commands return?
```
ls -l /var/packages/SynologyDrive
grep 'volume' /var/packages/SynologyDrive/etc/sharesync/daemon.conf
grep 'volume' /var/packages/SynologyDrive/etc/sharesync/monitor.conf
grep 'volume' /var/packages/SynologyDrive/etc/sharesync/service.conf
readlink /var/packages/SynologyDrive/etc/repo
du -sh /volume1/@synologydrive
du -sh /volume2/@synologydrive
du -sh /volume1/@SynologyDriveShareSync
du -sh /volume2/@SynologyDriveShareSync
```

**Reply by TorbenSchreiter** [reply]
All of them seem to have `/volume2` everywhere except for these two:

```
user123@GLHV3:/volume2$ readlink /var/packages/SynologyDrive/etc/repo
/volume1/@synologydrive/@sync

user123@GLHV3:/volume2$ du -sh /volume1/@synologydrive
64M     /volume1/@synologydrive
```

**Reply by 007revad** [reply]
What about the sizes from:
```
du -sh /volume2/@synologydrive
du -sh /volume1/@synologydrive.old
```

**Reply by TorbenSchreiter** [reply]
Attention: `/volume2` is btrfs while `/volume1` is ext4. I am pretty sure, a lot more than 19GB were copied!


```
user123@GLHV3:/volume2$ du -sh /volume2/@synologydrive
19G     /volume2/@synologydrive
```

```
user123@GLHV3:/volume2$ du -sh /volume1/@synologydrive.old
[..]
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/x/.4': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/x/.s': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/x/.y': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/x/.j': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/x/.B': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/x/.b': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/x/.1': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/x/.-': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/x/.5': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/x/._': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/x/.6': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/x/.0': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/w/.7': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/w/.K': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/w/.I': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/w/.8': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/w/.z': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/w/.g': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/w/.c': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/w/.E': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/w/.e': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/w/.V': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/w/.a': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/w/.A': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/w/.9': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/w/.4': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/w/.y': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/w/.b': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/w/.f': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/w/.5': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/w/.d': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/w/._': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/w/.W': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/w/.Q': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/w/.q': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/w/.h': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/w/.T': Permission denied
du: cannot access '/volume1/@synologydrive.old/@sync/delta/3/w/.6': Permission denied
591G    /volume1/@synologydrive.old
```

**Reply by 007revad** [reply]
Sorry, try:
```
sudo du -sh /volume1/@synologydrive.old
```
/volume2 being btrfs and /volume1 ext4 may be a problem

**Reply by TorbenSchreiter** [reply]
```
user123@GLHV3:/volume2$ sudo du -sh /volume1/@synologydrive.old
Password:
617G    /volume1/@synologydrive.old
```

**Reply by 007revad** [reply]
Stop Synology Drive Server then
```
sudo rm /var/packages/SynologyDrive/etc/repo
sudo ln -s "/volume2/@synologydrive/@sync" /var/packages/SynologyDrive/etc/repo
sudo rsync -aH --progress "/volume1/@synologydrive.old/" "/volume2/@synologydrive"
```
Then check with
```
readlink /var/packages/SynologyDrive/etc/repo
sudo du -sh /volume2/@synologydrive
```

**Reply by TorbenSchreiter** [reply]
Ok. it's 10pm here, now and it's going to take a while until copying will be done. Will leave Putty open and hope it'll finish alone. Will report back in the am. Thanks!

**Reply by TorbenSchreiter** [reply]
Good morning. At first it looked ok after the move was complete:
```
user123@GLHV3:/volume2$ readlink /var/packages/SynologyDrive/etc/repo
/volume2/@synologydrive/@sync
user123@GLHV3:/volume2$ sudo du -sh /volume2/@synologydrive
Password:
619G    /volume2/@synologydrive
```

I then started up the package, which showed up w/ default settings again. And afterwards 🫣:
```
user123@GLHV3:/volume1$ readlink /var/packages/SynologyDrive/etc/repo
/volume1/@synologydrive/@sync
```

**Reply by 007revad** [reply, marked as answer]
I just noticed on my Synology Drive is on volume 1, but "Synology Drive Admin > Settings" shows the database volume is volume 3.

Checking via SSH I see: 
1. /var/packages/SynologyDrive/etc/db-path.conf contains `db-vol=/volume3`  # The script does not edit this file
2. The "/var/packages/SynologyDrive/etc/repo" symlink points to `/volume3/@synologydrive/@sync`

Maybe the script should not edit the repo symlink and also not move the "@synologydrive" folder, as "Synology Drive Admin > Settings" can move the database.

Try this: 
1. Stop Synology Drive Server.
2. Then run these commands:
```
sudo rm /var/packages/SynologyDrive/etc/repo
sudo ln -s "/volume2/@synologydrive/@sync" /var/packages/SynologyDrive/etc/repo
sudo synosetkeyvalue /var/packages/SynologyDrive/etc/db-path.conf db-vol /volume2
```
3. Start Synology Drive Server.
4. Check the changes have not reverted back to /volume1
```
readlink /var/packages/SynologyDrive/etc/repo
synogetkeyvalue /var/packages/SynologyDrive/etc/db-path.conf db-vol
```

**Reply by TorbenSchreiter** [reply]
In step 2 I would replace `/volume 3` with `/volume2` (last line), correct?

**Reply by 007revad** [reply]
Yes.
```
sudo synosetkeyvalue /var/packages/SynologyDrive/etc/db-path.conf db-vol /volume2
```

**Reply by TorbenSchreiter** [reply]
Yes, that might have done the trick! Settings are back and changes have not yet reverted back. Also the UI dropdown in the Drive Admin Console now points to "Volume 2" (without me changing it):
```
user123@GLHV3:/volume1$ readlink /var/packages/SynologyDrive/etc/repo
/volume2/@synologydrive/@sync
user123@GLHV3:/volume1$ synogetkeyvalue /var/packages/SynologyDrive/etc/db-path.conf db-vol
/volume2
```

My desktop Drive client seems to have connected and has successfully synced new files that I had created locally in the meantime while the Synology Drive Server was down over the course of the last 2 days. Mobile clients connecting flawlessly as well.

Looks all good to me for now. What issues were you potentially expecting above, when you said migrating the package from ext4 to btrfs might cause problems? Something specific to watch out for from now?

Many thanks again! That was a great help!

**Reply by 007revad** [reply]
Awesome!

I'm going to change the script so it does **not** move `@synologydrive` or edit the symlink. That way people can use Synology Drive Admin to move the database.

**Reply by TorbenSchreiter** [reply]
That should also solve the aborted SSH session root cause for these monstrous size Synology Drive DBs...
